### PR TITLE
`azurerm_orchestrated_virtual_machine_scale_set` - add extension support

### PR DESCRIPTION
**Pull request by WodansSon**
Adding support for VMSS Flex VM extensions. This PR needs to wait for the service team Hotfix for concurrent extension provisioning to be deployed to all regions before it can be merged.

### CODE REVIEW

**Comment by WodansSon** [comment]
image

File: internal/services/compute/orchestrated_virtual_machine_scale_set.go [hunk 1]
```diff
@@ -135,73 +135,73 @@ func OrchestratedVirtualMachineScaleSetLinuxConfigurationSchema() *pluginsdk.Sch
...
+					ValidateFunc: validation.StringIsNotEmpty,
+				},
+
+				"type_handler_version": {
+					Type:         pluginsdk.TypeString,
+					Required:     true,
+					ValidateFunc: validation.StringIsNotEmpty,
+				},
+
+				"auto_upgrade_minor_version": {
```
Comment (katbyte) [inline internal/services/compute/orchestrated_virtual_machine_scale_set.go:169]: could we add enabled here?

File: internal/services/compute/orchestrated_virtual_machine_scale_set.go [hunk 2]
```diff
@@ -135,73 +135,73 @@ func OrchestratedVirtualMachineScaleSetLinuxConfigurationSchema() *pluginsdk.Sch
...
+					Type:     pluginsdk.TypeBool,
+					Optional: true,
+					Default:  true,
+				},
+
+				"force_update_tag": {
```
Comment (katbyte) [inline internal/services/compute/orchestrated_virtual_machine_scale_set.go:175]: could we change this? I think somewhere else it became "force update rescue when changed" or something - this does make much sense as is

File: internal/services/compute/orchestrated_virtual_machine_scale_set.go [hunk 3]
```diff
@@ -135,73 +135,73 @@ func OrchestratedVirtualMachineScaleSetLinuxConfigurationSchema() *pluginsdk.Sch
...
+				},
+
+				"protected_settings": {
+					Type:         pluginsdk.TypeString,
+					Optional:     true,
+					Sensitive:    true,
+					ValidateFunc: validation.StringIsJSON,
+				},
+
+				"provision_after_extensions": {
```
Comment (katbyte) [inline internal/services/compute/orchestrated_virtual_machine_scale_set.go:187]: provision what? should this be provision_this_after_extensions

File: internal/services/compute/orchestrated_virtual_machine_scale_set.go [hunk 4]
```diff
@@ -135,73 +135,73 @@ func OrchestratedVirtualMachineScaleSetLinuxConfigurationSchema() *pluginsdk.Sch
...
+					Type:     pluginsdk.TypeList,
+					Optional: true,
+					Elem: &pluginsdk.Schema{
+						Type: pluginsdk.TypeString,
+					},
```
Comment (katbyte) [inline internal/services/compute/orchestrated_virtual_machine_scale_set.go:192]: should we be validating this?

**Review by katbyte — CHANGES_REQUESTED** [review]

File: internal/services/compute/orchestrated_virtual_machine_scale_set.go (hunk 1 above)
Comment (WodansSon) [reply internal/services/compute/orchestrated_virtual_machine_scale_set.go:169]: Fixed.

File: internal/services/compute/orchestrated_virtual_machine_scale_set.go (hunk 2 above)
Comment (WodansSon) [reply internal/services/compute/orchestrated_virtual_machine_scale_set.go:175]: Fixed... went with `force_extension_execution_on_change`

File: internal/services/compute/orchestrated_virtual_machine_scale_set.go (hunk 3 above)
Comment (WodansSon) [reply internal/services/compute/orchestrated_virtual_machine_scale_set.go:187]: Fixed... I updated it to be `extensions_to_provision_after_vm_creation` which is exactly what this field does.

File: internal/services/compute/orchestrated_virtual_machine_scale_set.go (hunk 4 above)
Comment (WodansSon) [reply internal/services/compute/orchestrated_virtual_machine_scale_set.go:192]: Fixed.

**Comment by WodansSon** [comment]
image

**Review by katbyte — APPROVED** [review]
LGTM 🙏

**Pull request merged**